CRM/Admin/Form/RelationshipType.php

```diff
@@ -157,20 +156,22 @@ public function postProcess() {
       $params['contact_type_a'] = $cTypeA[0];
       $params['contact_type_b'] = $cTypeB[0];
 
-      $params['contact_sub_type_a'] = $cTypeA[1] ? $cTypeA[1] : 'NULL';
-      $params['contact_sub_type_b'] = $cTypeB[1] ? $cTypeB[1] : 'NULL';
+      $params['contact_sub_type_a'] = $cTypeA[1] ? $cTypeA[1] : 'null';
+      $params['contact_sub_type_b'] = $cTypeB[1] ? $cTypeB[1] : 'null';
 
-      // if label B to A is blank, insert the value label A to B for it
-      if (!strlen(trim(CRM_Utils_Array::value('name_b_a', $params)))) {
-        $params['name_b_a'] = CRM_Utils_Array::value('name_a_b', $params);
-      }
       if (!strlen(trim(CRM_Utils_Array::value('label_b_a', $params)))) {
         $params['label_b_a'] = CRM_Utils_Array::value('label_a_b', $params);
       }
 
-      $result = CRM_Contact_BAO_RelationshipType::add($params);
+      if (empty($params['id'])) {
+        // Set name on created but don't update on update as the machine name is not exposed.
+        $params['name_b_a'] = CRM_Utils_String::munge($params['label_b_a']);
+        $params['name_a_b'] = CRM_Utils_String::munge($params['label_a_b']);
+      }
+
+      $result = civicrm_api3('RelationshipType', 'create', $params);
 
-      $this->ajaxResponse['relationshipType'] = $result->toArray();
+      $this->ajaxResponse['relationshipType'] = $result['values'];
```

Comment:
Do we need an `isset` or `CRM_Utils_Array::value` on the $result['values'] as it may not be set if RelationshipType.create fails for some reason and we'll end up with a PHP notice.

Reply:
It would throw an exception if there were an error & the form would handle that with a messy message - but that's OK since it is not an expected exception